.github/workflows/make.yml

```diff
@@ -0,0 +1,117 @@
+name: Build Emulation Cores
+
+on:
+  pull_request:
+    branches: [ "master" ]
+  push:
+    branches: [ "master" ]
+
+env:
+  CC: clang-18
+  CXX: clang++-18
+
+jobs:
+
+  build-waterbox:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Installing clang 18
+      run: wget https://apt.llvm.org/llvm.sh;
+          chmod u+x llvm.sh;
+          sudo ./llvm.sh 18;
+          clang-18 --version;    
+    - name: Getting submodule(s)
+      working-directory: ./waterbox/
+      run: git submodule update --init --recursive musl
+    - name: Building musl
+      working-directory: ./waterbox/musl
+      run: ./wbox_configure.sh;
+           ./wbox_build.sh;
+    - name: Building emulibc
+      working-directory: ./waterbox/emulibc
+      run: make
+    - name: Building libco
+      working-directory: ./waterbox/libco
+      run: make
+    - name: Building libcxx
+      working-directory: ./waterbox/libcxx
+      run: ./do-everything.sh
+    - name: Building nyma
+      working-directory: ./waterbox/nyma
+      run: ./build-and-install-zlib.sh
+    - name: Uploading compiled waterbox
+      uses: actions/upload-artifact@v4
+      with:
+        name: compiled-waterbox
+        path: |
+          waterbox/libco
+          waterbox/musl
+          waterbox/emulibc
+          waterbox/libcxx
+          waterbox/sysroot
+  
+  build-waterboxed-cores:
+    runs-on: ubuntu-latest
+    needs: build-waterbox
+    steps:
+    - uses: actions/checkout@v4
+    - name: Getting submodule(s)
+      working-directory: ./waterbox/
+      run: git submodule update --init --recursive snes9x;
+           git submodule update --init --recursive melon/melonDS;
+           git submodule update --init --recursive nyma/mednafen;
+           git submodule update --init --recursive ares64/ares/thirdparty/angrylion-rdp;
+           git submodule update --init --recursive gpgx/Genesis-Plus-GX;
```

Comment:
You don't actually need the init these recursively. ~~I don't think this is explicitly communicated anywhere and it probably should be~~ (UPDATE: have clarified this in the readme), but for these cores no recursive submodules are necessary. I think the only exception would be the `encore/encore` submodule in the repository root. Removing the `--recursive` calls here would help for snes9x specifically, as it has a number of heavy submodule dependencies that are not necessary for our waterbox build.

Additionally, as per the waterbox readme, to build all waterbox cores correctly the `submodules/sameboy/libsameboy` submodule is also required.